### PR TITLE
adds additional pools object

### PR DESCRIPTION
**Pull request by volatilemolotov**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: .github/gh-actions-self-hosted-runners/arc/gke.tf [hunk 1]
```diff
@@ -26,26 +26,70 @@ resource "google_container_cluster" "actions-runner-gke" {
...
    }
+   initial_node_count = var.main_runner.min_node_count
   management {
     auto_repair  = "true"
     auto_upgrade = "true"
    }
   node_config {
-    machine_type    = var.machine_type
+    disk_size_gb = var.main_runner.disk_size_gb
+    machine_type    = var.main_runner.machine_type
```
Comment (andreydevyatkin) [inline .github/gh-actions-self-hosted-runners/arc/gke.tf:44]: nit: formatting

File: .github/gh-actions-self-hosted-runners/arc/gke.tf [hunk 2]
```diff
@@ -26,26 +26,70 @@ resource "google_container_cluster" "actions-runner-gke" {
...
+      for_each = each.value.enable_taint == true ? [1] : []
+      content {
+        key = "runner-pool"
+        value = each.value.name
+        effect = "NO_SCHEDULE"
+        }
+    }
+   }
+
+}
```
Comment (andreydevyatkin) [inline .github/gh-actions-self-hosted-runners/arc/gke.tf:90]: nit: formatting

File: .github/gh-actions-self-hosted-runners/arc/environments/beam.env [hunk 3]
```diff
@@ -30,9 +28,33 @@ github_app_id_secret_name = "gh-app_id"
...
+    machine_type = "e2-standard-2"
+    min_node_count = "1"
+    max_node_count = "10"
+    min_replicas = "1"
+    max_replicas = "10"
+    webhook_scaling = "true"
+    requests = {
+        cpu = 1500m
+    }
+    name = "small-runners"
```
Comment (andreydevyatkin) [inline .github/gh-actions-self-hosted-runners/arc/environments/beam.env:57]: duplicate name

File: .github/gh-actions-self-hosted-runners/arc/environments/beam.env [hunk 4]
```diff
@@ -30,9 +28,33 @@ github_app_id_secret_name = "gh-app_id"
...
+additional_runner_pools = [{
+    name = "small-runner"
+    machine_type = "e2-standard-2"
+    min_node_count = "1"
+    max_node_count = "10"
+    min_replicas = "1"
+    max_replicas = "10"
+    webhook_scaling = "true"
+    requests = {
+        cpu = 1500m
```
Comment (andreydevyatkin) [inline .github/gh-actions-self-hosted-runners/arc/environments/beam.env:55]: it make sense to also specify the memory value

**Review by andreydevyatkin — COMMENTED** [review]
see the inline comments

File: .github/gh-actions-self-hosted-runners/arc/README.md [hunk 5]
```diff
@@ -48,11 +55,34 @@ github_app_id_secret_name = "app_id_secret_name"              # Google secret na
...
+enable_selector = "true"                  # Enables NodeSelector, forcing runners to this pool
+enable_taint = "true"                     # Enables Taints. Prevents other runner pods to run in this pool.
+requests = {                              # K8s cpu and memory requests
+  cpu = "500m"                            #
+  memory = "500mi"}                       #
+limits = {                                # K8s cpu and memory limits
+    cpu = "2"                             #
+    memory = "2Gi"}}]                     #
+
+```
```
[the remark this answers is not in the document]
Comment (volatilemolotov) [reply .github/gh-actions-self-hosted-runners/arc/README.md:82]: Formatting is fine once you open it in codeview.

File: .github/gh-actions-self-hosted-runners/arc/gke.tf (hunk 1 above)
Comment (volatilemolotov) [reply .github/gh-actions-self-hosted-runners/arc/gke.tf]: fixed

File: .github/gh-actions-self-hosted-runners/arc/environments/beam.env (hunk 3 above)
Comment (volatilemolotov) [reply .github/gh-actions-self-hosted-runners/arc/environments/beam.env]: fixed

File: .github/gh-actions-self-hosted-runners/arc/environments/beam.env (hunk 4 above)
Comment (volatilemolotov) [reply .github/gh-actions-self-hosted-runners/arc/environments/beam.env:55]: At this point no. We use cpu to achieve 1 per node so memory is not important in this context.

**Pull request closed without merging**